examples/bq-email-exports/main.py

```diff
@@ -0,0 +1,181 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# using SendGrid's Python Library
+# https://github.com/sendgrid/sendgrid-python
+
+import os
+import time
+import datetime
+from sendgrid import SendGridAPIClient
+from sendgrid.helpers.mail import Mail
+from google.cloud import bigquery, storage
+from google.auth import iam, default
+from google.auth.transport import requests
+from google.oauth2 import service_account
+
+
+def credentials():
+    """Gets credentials to authenticate Google APIs.
+
+    Args:
+        None
+
+    Returns:
+        Credentials to authenticate the API.
+    """
+    # Get Application Default Credentials if running in Cloud Functions
+    if os.getenv("IS_LOCAL") is None:
```

Comment:
A few things:
1. Can we handle this more elegantly in one place? without requiring user to set IS_LOCAL just use try / except or use a method of getting credentials that already handles both execution environments. @danieldeleo @tswast 
1. In general don't try to make one entry point for GCF and a CLI tool (you can dry up the common implementation into a module (separate python file where you have a function that accepts credentials ) and provide a small GCF wrapper and CLI wrappers (that takes useful runtime args with argparse, grab the local credentials and call the common module))

Reply:
@jaketf I think this is a great idea. Particularly I think having 2 entry points, i.e the main module that GCF uses and a local script that takes in local credentials will solve this problem well.  

But now that we have moved to a TF solution does it make sense to have the local part? 
Initially this local support was primarily just for quick iteration and testing but I assume this would not be needed in a real world use case. If you think it's a valuable addition, happy to add this.